draft-ietf-ppm-dap.md

```diff
@@ -1231,23 +1235,30 @@ struct {
   ReportID report_id;
   PrepareStepResult prepare_step_result;
   select (PrepareStep.prepare_step_result) {
-    case continued: opaque prep_msg<0..2^32-1>; /* VDAF preparation message */
+    case continued: opaque prep_msg<0..2^32-1>; /* VDAF preparation share or message */
```

Comment:
IIUC, `ReportInitialization.init_prep_step` and `ReportInitialization.stepped_prepare_step` should always be in the same state (that is, both continued, both finished, or both failed), right? If so, then it's unfortunate that this representation allows constructing invalid combinations of `init_prep_step` and `stepped_prepare_step`. How about:

```suggestion
    case continued: {
        opaque init_prep_msg<0..2^32-1>; /* VDAF preparation share or message */
        opaque stepped_prepare_msg<0..2^32-1>; /* VDAF preparation share or message */
    }
```
(or substitute something that is valid in the presentation language)

And then `struct ReportInitialization` would contain a single `PrepareStep`. This would also avoid transmitting the same `report_id` value twice in each `ReportInitialization`.

Reply:
`init_prep_step` and `stepped_prepare_step` won't necessarily be in the same state: see the Input Share Preparation section. `init_prep_step` includes the helper's first call to `prep_next`, `stepped_prepare_step` includes the helper's second call to `prep_next`.

(note that the use of `PrepareStep` in this way does repeat the `report_id` unnecessarily; `PrepareStep` is still used in a number of places that need to include a `report_id`. It'd be easy enough to factor the messages into `PrepareStep`-without-report-ID & `PrepareStep`-with-report-ID, but I'll do so once we're happy with the general approach.)